Client/Application/AppDelegate.swift

```diff
@@ -81,6 +82,23 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         return true
     }
 
+    func application(application: UIApplication, openURL url: NSURL, sourceApplication: String?, annotation: AnyObject?) -> Bool {
+        if url.scheme == "firefox" {
+            if url.lastPathComponent == "open-url" {
+                let query = url.query
```

Comment:
You're better off using if let here rather than checking for != nil
`if let query = url.query {
  ...
}`